src/async_impl/client.rs

```diff
@@ -129,6 +129,24 @@ impl Default for ClientBuilder {
     }
 }
 
+trait TlsConfig {
+    fn to_tls_backend(self) -> crate::tls::TlsBackend;
+}
+
+#[cfg(feature = "native-tls")]
+impl TlsConfig for native_tls_crate::TlsConnector {
+    fn to_tls_backend(self) {
+        crate::tls::TlsBackend::BuiltNativeTls(tls)
+    }
+}
+
+#[cfg(feature = "__rustls")]
+impl TlsConfig for native_tls_crate::ClientBuilder {
```

Comment:
This seems incorrect. You probably wanted `rustls::ClientConfig`.

Reply:
This is correct. See removed line 1264 and following:
```
            if let Some(conn) =		
                (&mut tls as &mut dyn Any).downcast_mut::<Option<rustls::ClientConfig>>()		
            {		
                let tls = conn.take().expect("is definitely Some");		
                let tls = crate::tls::TlsBackend::BuiltRustls(tls);		
                self.config.tls = tls;		
                return self;		
            }		
        }
 ```
 This impl replaces that downcast and is only used in that function.

Reply:
Whoops, looks like you're right. Let me fix that and build again with the appropriate features turned on.